apps/bfd-model/bfd-model-rif/src/main/resources/db/scripts/Query_synthetic_id_ranges.sql

```diff
@@ -0,0 +1,102 @@
+-- Find the beneficiaries.bene_id range to avoid collisions with synthetic data.
+WITH ids AS (
+  SELECT
+    CAST(bene_id AS numeric) AS id
+  FROM beneficiaries
+  ORDER BY CAST(bene_id AS numeric)
+),
+groupings AS (
+  SELECT
+    ROW_NUMBER() OVER (ORDER BY id) - id AS grouping,
+    id
+  FROM ids
+)
+SELECT MIN(id) AS grouping_start,
+       MAX(id) AS grouping_end
+FROM groupings
+GROUP BY grouping
+ORDER BY MIN(id);
+
+-- Find  min claim Ids for all claims except Part D Events.
+-- This serves as an end range to avoid collisions with future synthetic data batches.
+select min(id) from (
+	select CAST(clm_id AS numeric) id 
+	from carrier_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from dme_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from hha_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from hospice_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from inpatient_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from outpatient_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_id AS numeric) id 
+	from snf_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	) as ids;
+	
+-- Find  min Part D Event id range to avoid collisions with future synthetic data batches.
+select min(id) from (
+	select CAST(pde_id AS numeric) id 
+	from partd_events
+	where bene_id LIKE '-%' AND pde_id LIKE '-%'
+	) as ids;
+
+-- Find claim group id ranges.
+-- This query should be run before and after generating any synthetic data in prod SBX
+-- to gauge ensure how large the next batch of synthetic data can be without overlap.
+WITH ids AS (
+	select CAST(clm_grp_id AS numeric) id 
+	from carrier_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from dme_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from hha_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from hospice_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from inpatient_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from outpatient_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+	union
+	select CAST(clm_grp_id AS numeric) id 
+	from snf_claims
+	where bene_id LIKE '-%' AND clm_id LIKE '-%'
+  ORDER BY id
+),
+groupings AS (
+  SELECT
+    ROW_NUMBER() OVER (ORDER BY id) - id AS grouping,
+    id
+  FROM ids
+)
+SELECT MIN(id) AS grouping_start,
+       MAX(id) AS grouping_end
+FROM groupings
+GROUP BY grouping
+ORDER BY MIN(id);
```

Comment:
Please add the output in a comment for each of the 4 queries from running this against prod-sandbox after you load synthetic data into it so that it is up to date.